charts/argo-cd/templates/argocd-repo-server/deployment.yaml

```diff
@@ -44,24 +45,91 @@ spec:
         - entrypoint.sh
         args:
         - argocd-repo-server
-        {{- if or (and .Values.redis.enabled (not $redisHa.enabled)) (and $redisHa.enabled $redisHa.haproxy.enabled) }}
-        - --redis
-        - {{ template "argo-cd.redis.fullname" . }}:{{ .Values.redis.servicePort }}
-        {{- else if and .Values.externalRedis.host .Values.externalRedis.port }}
-        - --redis
-        - {{ .Values.externalRedis.host }}:{{ .Values.externalRedis.port }}
-        {{- end }}
-        - --logformat
-        - {{ .Values.repoServer.logFormat }}
-        - --loglevel
-        - {{ .Values.repoServer.logLevel }}
         {{- with .Values.repoServer.extraArgs }}
         {{- . | toYaml | nindent 8 }}
         {{- end }}
         {{- if .Values.repoServer.containerSecurityContext }}
         securityContext: {{- toYaml .Values.repoServer.containerSecurityContext | nindent 10 }}
         {{- end }}
         env:
+        - name: ARGOCD_RECONCILIATION_TIMEOUT
+          valueFrom:
+            configMapKeyRef:
+              name: argocd-cm
+              key: timeout.reconciliation
+              optional: true
+        - name: ARGOCD_REPO_SERVER_LOGFORMAT
+          valueFrom:
+            configMapKeyRef:
+              name: argocd-cmd-params-cm
+              key: reposerver.log.format
+              optional: true
+        - name: ARGOCD_REPO_SERVER_LOGLEVEL
+          valueFrom:
+            configMapKeyRef:
+              name: argocd-cmd-params-cm
+              key: reposerver.log.level
```

Comment:
@ArgTang Log level / format is now controlled via config map. Just provide proper key in params section for given component.

Reply:
ah you are right. thanks for the correction :)